src/ol_infrastructure/infrastructure/consul/__main__.py

```diff
@@ -400,6 +416,9 @@ def cloud_init_userdata(
             init_dict["retry_join_wan"],
             consul_dns_name,
             init_dict["pulumi_password"],
+            init_dict["grafana_api_key"],
+            init_dict["grafana_prometheus_user"],
+            init_dict["grafana_loki_user"],
```

Comment:
These will need to be added to the `Output.all` parameters and pulled from the stack configs.

Reply:
You should be able to avoid having to pass them through the lambda by using the `read_yaml_secrets` method inside the cloud-init function. For example: https://github.com/mitodl/ol-infrastructure/blob/maple_ami/src/ol_infrastructure/applications/edxapp/__main__.py#L934-L936

Reply:
How is that grafana yaml file generated?

Reply:
I created it using SOPS and checked it into the repository for the CI, QA, and Production environments.

Reply:
> These will need to be added to the `Output.all` parameters and pulled from the stack configs.

By stack configs are u referring to the grafana sops file where the creds are written to?  In looking at the code, unlike the vault one below, I can't use `grafana_credentials` as the Output function is outside the cloud-init one, so do we make another call to `read_yaml_secrets`? I'm also not very clear on what purpose this `Output.all` serves here.